src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -118,156 +118,174 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
     int sqlType = metadata.getColumnType(col);
 
     String defaultValue = whitelistDefaultValue(sqlType, columnDefaults.get(name));
-    boolean hasDefault = (defaultValue != null);
+    boolean hasDefault = false;
+    boolean defaultIsNull = false;
+    if (defaultValue != null) {
+      hasDefault = true;
+      defaultIsNull = defaultValue.equalsIgnoreCase("null");
+    }
+
 
     boolean optional = false;
     if (metadata.isNullable(col) == ResultSetMetaData.columnNullable ||
         metadata.isNullable(col) == ResultSetMetaData.columnNullableUnknown) {
       optional = true;
     }
 
-    Schema schema;
     switch (sqlType) {
       case Types.NULL: {
         log.warn("JDBC type {} not currently supported", sqlType);
         break;
       }
 
       case Types.BOOLEAN: {
+        SchemaBuilder fieldBuilder = SchemaBuilder.bool();
+        if (optional) {
+          fieldBuilder.optional();
+        }
         if (hasDefault) {
-          boolean castedDefault = Boolean.parseBoolean(defaultValue);
-          schema = SchemaBuilder.bool().defaultValue(castedDefault).build();
-        } else {
-          schema = (optional) ? Schema.OPTIONAL_BOOLEAN_SCHEMA : Schema.BOOLEAN_SCHEMA;
+          Boolean parsedDefault = defaultIsNull ? null : Boolean.parseBoolean(defaultValue);
```

Comment:
I also think it's totally wack that there is a distinction between 'optional' columns and non-optional, nullable with a default of 'null'. It's bananas.